libcxx/include/__expected/expected.h

```diff
@@ -1139,7 +1140,11 @@ class expected : private __expected_base<_Tp, _Err> {
 
   // [expected.object.eq], equality operators
   template <class _T2, class _E2>
-    requires(!is_void_v<_T2>)
+    requires(!is_void_v<_T2> &&
+             requires(const _Tp& __tp, const _T2& __t2, const _Err& __err, const _E2& __e2) {
+               { __tp == __t2 } -> convertible_to<bool>;
```

Comment:
The `convertible_to` concept is stricter than the plain "(implicitly) convertible to". The latter can be expressed by `__is_core_convertible` in `<__type_traits/is_core_convertible.h>`.

Reply:
I guess we can have the following.

<details><summary>Internal concept <code>__core_convertible_to</code> in the <code><__concepts/core_convertible_to.h></code> internal header</summary>
<p>

```C++
#ifndef _LIBCPP___CONCEPTS_CORE_CONVERTIBLE_TO_H
#define _LIBCPP___CONCEPTS_CORE_CONVERTIBLE_TO_H

#include <__config>

#if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
#  pragma GCC system_header
#endif

_LIBCPP_BEGIN_NAMESPACE_STD

#if _LIBCPP_STD_VER >= 20

// [conv.general]/3 says "E is convertible to T" whenever "T t=E;" is well-formed.
// We can't test for that, but we can test implicit convertibility by passing it
// to a function. Unlike std::convertible_to, __core_convertible_to doesn't test
// static_cast or handle cv void, while accepting move-only types.

template <class _Tp, class _Up>
concept __core_convertible_to = requires {
  // rejects function and array types which are adjusted to pointer types in parameter lists
  static_cast<_Up (*)()>(nullptr)();
  static_cast<void (*)(_Up)>(nullptr)(static_cast<_Tp (*)()>(nullptr)());
};

#endif

_LIBCPP_END_NAMESPACE_STD

#endif // _LIBCPP___CONCEPTS_CORE_CONVERTIBLE_TO_H
```

</p>
</details>

Reply:
Would you like to draft a patch for this header if you have time? I can rebase and use your header. This will ensure that you receive credit for your work! Otherwise, I can include it in this patch myself.

Reply:
It seems we used `__boolean_testable` to implement the constraints for `operator==` in `std::reference_wrapper`. (https://github.com/llvm/llvm-project/blob/main/libcxx/include/__functional/reference_wrapper.h#L75-L81)  
Should we consider making them consistent?